hip-lang-config.cmake.in

```diff
@@ -96,15 +159,37 @@ if (NOT HSA_HEADER)
   message (FATAL_ERROR "HSA header not found! ROCM_PATH environment not set")
 endif()
 
-get_filename_component(HIP_COMPILER_INSTALL_PATH ${CMAKE_HIP_COMPILER} DIRECTORY)
-file(GLOB HIP_CLANGRT_LIB_SEARCH_PATHS "${HIP_COMPILER_INSTALL_PATH}/../lib/clang/*/lib/*")
-find_library(CLANGRT_BUILTINS
-    NAMES
-      clang_rt.builtins
-      clang_rt.builtins-x86_64
-    PATHS
-      ${HIP_CLANGRT_LIB_SEARCH_PATHS}
-      ${HIP_CLANG_INCLUDE_PATH}/../lib/linux)
+# CLANGRT LIB Search Path setting based on valid LLVM Path found
+if( LLVM_FOUND )
+  # IF VALID LLVM Package successfully found using find_dependency(LLVM)
+  # Use binary tool option -print-libgcc-file-name --rtlib=compiler-rt
+  # To fetch the compiler rt library file name.
+  execute_process(
+    COMMAND ${CLANG_TOOL_BINARY} -print-libgcc-file-name --rtlib=compiler-rt
+    OUTPUT_VARIABLE CLANGRT_BUILTINS
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    RESULT_VARIABLE CLANGRT_BUILTINS_FETCH_EXIT_CODE)
+else()
```

Comment:
Remove this `else` branch, and also remove the `if( LLVM_FOUND )` as well. We should always call the hip compiler to get the compiler-rt path. We should probably do something similar with hip-config.cmake.